filebeat/input/kafka/input.go

```diff
@@ -308,13 +311,13 @@ func (h *groupHandler) ack(message *sarama.ConsumerMessage) {
 
 func (h *groupHandler) ConsumeClaim(session sarama.ConsumerGroupSession, claim sarama.ConsumerGroupClaim) error {
 	reader := h.createReader(claim)
-	parser := h.parsers.Create(reader)
-	for h.session.Context().Err() == nil {
+	parser := h.parsers.Create(h.ctx, reader)
+	for errors.Is(h.session.Context().Err(), nil) {
```

Comment:
```suggestion
	for h.session.Context().Err() == nil {
```

errorlint [should not complain](https://github.com/polyfloyd/go-errorlint/blob/a0c208956a365f0f594df070d05d5359f1492648/errorlint/lint.go#L152-L154) about comparison of errors to nil.